web/src/composebox_typeahead.js

```diff
@@ -151,38 +152,70 @@ export function should_enter_send(e) {
     return this_enter_sends;
 }
 
+function handle_bulleting_or_numbering($textarea, e) {
+    // handles automatic insertion or removal of bulleting or numbering
+    const before_text = split_at_cursor($textarea.val(), $textarea)[0];
+    const previous_line = list_util.get_last_line(before_text);
+    let to_append = "";
+    // if previous line was bulleted automatically add a bullet to the new line
+    if (list_util.is_bulleted(previous_line)) {
+        // if previous line had only bullet, remove it and stay on the same line
+        if (list_util.strip_bullet(previous_line) === "") {
+            // the below line replaces the last 2 characters in the textarea before
+            // the cursor - the bullet syntax - with an empty string
+            $textarea.range($textarea.caret() - 2, $textarea.caret()).range("");
```

Comment:
This doesn't use the `text-field-edit` modules and thus this feature breaks undo.

Reply:
Changed to use [`setSelectionRange`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setSelectionRange) followed by `text-field-edit`'s `insert` function, which retains undo-ability